test/e2e/common/node/pod_resize.go

```diff
@@ -1012,6 +1013,565 @@ func doPodResizeErrorTests(f *framework.Framework) {
+			containers: []e2epod.ResizableContainerInfo{
+				{
+					Name:      "c1",
+					Resources: &e2epod.ContainerResources{CPUReq: "100m", CPULim: "200m", MemReq: "200Mi", MemLim: "300Mi"},
+					CPUPolicy: &noRestart,
+					MemPolicy: &noRestart,
+				},
+			},
+			containerPatches: [][]v1.Container{
+				{
+					{
+						Name: "c1",
+						Resources: v1.ResourceRequirements{
+							Limits: v1.ResourceList{
+								v1.ResourceCPU:    resource.MustParse("250m"),
+								v1.ResourceMemory: resource.MustParse("350Mi"),
+							},
+						},
+					},
+				},
+				{
+					{
+						Name: "c1",
+						Resources: v1.ResourceRequirements{
+							Limits: v1.ResourceList{
+								v1.ResourceCPU:    resource.MustParse("300m"),
+								v1.ResourceMemory: resource.MustParse("400Mi"),
+							},
+						},
+					},
+				},
+			},
+			intervals: []time.Duration{1 * time.Second},
+			expected: [][]e2epod.ResizableContainerInfo{
+				{},
+				{{
+					Name:         "c1",
+					Resources:    &e2epod.ContainerResources{CPUReq: "100m", CPULim: "300m", MemReq: "200Mi", MemLim: "400Mi"},
+					CPUPolicy:    &noRestart,
+					MemPolicy:    &noRestart,
+					RestartCount: 0,
+				}},
+			},
+		},
+		{
+			isSynchronous: false,
+			name:          "Resize only requests without restarting twice asynchronously",
+			containers: []e2epod.ResizableContainerInfo{
+				{
+					Name:      "c1",
+					Resources: &e2epod.ContainerResources{CPUReq: "100m", CPULim: "300m", MemReq: "200Mi", MemLim: "400Mi"},
+					CPUPolicy: &noRestart,
+					MemPolicy: &noRestart,
+				},
+			},
+			containerPatches: [][]v1.Container{
+				{
+					{
+						Name: "c1",
+						Resources: v1.ResourceRequirements{
+							Requests: v1.ResourceList{
+								v1.ResourceCPU:    resource.MustParse("150m"),
+								v1.ResourceMemory: resource.MustParse("250Mi"),
+							},
+						},
+					},
+				},
+				{
+					{
+						Name: "c1",
+						Resources: v1.ResourceRequirements{
+							Requests: v1.ResourceList{
+								v1.ResourceCPU:    resource.MustParse("200m"),
+								v1.ResourceMemory: resource.MustParse("300Mi"),
+							},
+						},
+					},
+				},
+			},
+			intervals: []time.Duration{1 * time.Second},
+			expected: [][]e2epod.ResizableContainerInfo{
+				{},
+				{{
+					Name:         "c1",
+					Resources:    &e2epod.ContainerResources{CPUReq: "200m", CPULim: "300m", MemReq: "300Mi", MemLim: "400Mi"},
+					CPUPolicy:    &noRestart,
+					MemPolicy:    &noRestart,
+					RestartCount: 0,
+				}},
+			},
+		},
+	}
+
+	timeouts := framework.NewTimeoutContext()
+
+	for idx := range tests {
+		tc := tests[idx]
+		ginkgo.It(tc.name, func(ctx context.Context) {
+			var testPod, patchedPod *v1.Pod
+			var pErr error
+
+			tStamp := strconv.Itoa(time.Now().Nanosecond())
+			e2epod.InitDefaultResizePolicy(tc.containers)
+			for _, e := range tc.expected {
+				if len(e) > 0 {
+					e2epod.InitDefaultResizePolicy(e)
+				}
```

Comment:
I don't think we need to check the length. `InitDefaultResizePolicy` will handle slices with length 0.

```suggestion
				e2epod.InitDefaultResizePolicy(e)
```